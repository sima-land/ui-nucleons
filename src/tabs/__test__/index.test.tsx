import { render } from '@testing-library/react';
import { Tabs } from '..';

describe('<Tabs />', () => {
  it('should renders correctly', async () => {
    const { container, findAllByTestId } = render(
      <Tabs>
        <Tabs.Item name='One' />
        <Tabs.Item name='Two' disabled />
        <Tabs.Item name='Three' selected />
        <Tabs.Item name='Four' />
      </Tabs>,
    );

    expect(container).toMatchSnapshot();
    expect(await findAllByTestId('tabs')).toHaveLength(1);
    expect(await findAllByTestId('tab')).toHaveLength(4);
  });

  it('items should handle children', async () => {
    const { container } = render(
      <Tabs>
        <Tabs.Item>One</Tabs.Item>
        <Tabs.Item>Two</Tabs.Item>
        <Tabs.Item>Three</Tabs.Item>
      </Tabs>,
    );

    expect(container.textContent).toBe('OneTwoThree');
  });
});
