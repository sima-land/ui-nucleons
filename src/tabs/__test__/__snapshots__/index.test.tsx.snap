// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs should renders correctly 1`] = `
<div>
  <ul
    class="root view-clean gap-m"
    data-testid="tabs"
  >
    <li
      class="item"
      data-testid="tab"
    >
      One
    </li>
    <li
      class="item disabled"
      data-testid="tab"
    >
      Two
    </li>
    <li
      class="item selected"
      data-testid="tab"
    >
      Three
    </li>
    <li
      class="item"
      data-testid="tab"
    >
      Four
    </li>
  </ul>
</div>
`;
